src/coreclr/jit/codegencommon.cpp

```diff
@@ -6131,6 +6143,14 @@ void CodeGen::genFnProlog()
         intRegState.rsCalleeRegArgMaskLiveIn &= ~RBM_SECRET_STUB_PARAM;
     }
 
+#ifdef SWIFT_SUPPORT
+    if ((compiler->lvaSwiftSelfArg != BAD_VAR_NUM) && ((intRegState.rsCalleeRegArgMaskLiveIn & RBM_SWIFT_SELF) != 0))
```

Comment:
Are there situations where `lvaSwiftSelfArg` is valid, but the reg mask doesn't include the Swift self reg? If not, maybe we should assert that the reg mask includes it if `lvaSwiftSelfArg` is set.

Reply:
Yes, if `lvaSwiftSelfArg` is dead on entry then it shouldn't be in the live-in set.